test/integration/smoke/test_activate_template.py

```diff
@@ -0,0 +1,214 @@
+        cls._cleanup = []
+
+        cls.services = cls.testClient.getParsedTestDataConfig()
+        cls.unsupportedHypervisor = False
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            admin=True,
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.account)
+        cls.user = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.user)
+
+        cls.listTemplatesCmd = listTemplates.listTemplatesCmd()
+        cls.listTemplatesCmd.templatefilter= 'system'
+        # Save the current system template
+        cls.templates = cls.apiclient.listTemplates(cls.listTemplatesCmd)
+        cls.firstRun = True
+    
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            if cls.firstRun:
+                # Save currently active template
+                activeTemplate = cls.apiclient.listTemplates(cls.listTemplatesCmd)[0]
+                cmd = deleteTemplate.deleteTemplateCmd()
+                cls.debug(activeTemplate)
+                cmd.id = activeTemplate.id
+                cmd.zoneid = cls.zone.id
+
+                # Activate initial system template
+                for temp in cls.templates:
+                    cmd = activateSystemVMTemplate.activateSystemVMTemplateCmd()
+                    cmd.id = temp.id
+                    cls.apiclient.activateSystemVMTemplate(cmd)
+
+                # Remove test template
+                cls.apiclient.deleteTemplate(cmd)
+                cls.firstRun = False
+
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        # Default URL
+        self.getDefaultURLCmd = getSystemVMTemplateDefaultUrl.getSystemVMTemplateDefaultUrlCmd()
+        self.getDefaultURLCmd.hypervisor = self.hypervisor
+        urlResponse = self.apiclient.getSystemVMTemplateDefaultUrl(self.getDefaultURLCmd)
+        
+        # Official cloudstack system vm template
+        self.test_template = registerTemplate.registerTemplateCmd()
+        self.test_template.hypervisor = self.hypervisor
+        self.test_template.zoneid = self.zone.id
+        self.test_template.name = 'test-system-' + self.hypervisor + '-4.11.3'
+        self.test_template.displaytext = 'test-system-' + self.hypervisor + '-4.11.3'
+        self.test_template.url = urlResponse.url.url #"http://download.cloudstack.org/systemvm/4.11/systemvmtemplate-4.11.3-kvm.qcow2.bz2"
+        self.test_template.format = "QCOW2"
```

Comment:
format should be hypervisor specific also

Reply:
I think it would be good to return the template file type as part of the getsystemVMTemplatedefaultURL call. I will add this.

Reply:
Done